### PR TITLE
Export users to CSV format

### PR DESCRIPTION
**Pull request by dovganych**
**CHECK LIST**
- [x] specs added
- [X] specs passed
- [X] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [X] PR is reviewed manually again (to make sure you have 100% ready code)
- [X] PR doesn't have conflicts with main branch
- [X] I've checked new feature as logged in and logged out user if needed
- [ ] "let!" is used in specs
- [ ] screenshot of rails_best_practices and other tools added
- [X] screenshot of feature is added if task has UI

new_btn

downloaded file

users_csv_v2

### CODE REVIEW

File: app/views/account/users/index.html.slim [hunk 1]
```diff
@@ -11,7 +11,11 @@
             = render_breadcrumbs({'Dashboard' => account_root_url,
                   'Users' => nil})
       .content-header-right.text-md-right.col-md-6.col-12
-        = link_to 'New user', new_account_user_path, class: 'btn btn-primary'
+        .row
```
Comment (igorkasyanchuk) [inline app/views/account/users/index.html.slim]: use btn group

File: app/controllers/account/users_controller.rb [hunk 2]
```diff
@@ -58,6 +58,10 @@ def destroy
     redirect_to account_users_path, flash: { success: 'User deleted!' }
   end
 
+  def download
+    send_data collection.to_comma, filename: "Users #{Date.today}.csv"
```
Comment (igorkasyanchuk) [inline app/controllers/account/users_controller.rb]: disposition attachement

File: config/routes.rb [hunk 3]
```diff
@@ -29,6 +29,7 @@
     resources :users do
       post :impersonate, on: :member
       post :stop_impersonating, on: :collection
+      get  :download, on: :member
```
Comment (romaslmd) [inline config/routes.rb]: could you make some structure here by grouping, e.g.:
```
member do
  # ...
end

collection do
  # ...
end
```

File: app/views/account/users/index.html.slim [hunk 4]
```diff
@@ -11,7 +11,11 @@
...
+          .offset-md-4.col-md-1
+            = link_to 'Add new', new_account_user_path, class: 'btn btn-primary'
+          .offset-md-1.col-md-1
+            = link_to 'Users CSV', download_account_user_path(@users), class: 'btn btn-info'
```
Comment (romaslmd) [inline app/views/account/users/index.html.slim]: mb `@user` instead of `@users` ?

Comment (igorkasyanchuk) [reply app/views/account/users/index.html.slim]: Good catch Roman. This route doesn't require parameter at all

Comment (igorkasyanchuk) [reply app/views/account/users/index.html.slim]: Good catch Roman. This route doesn't require parameter at all

File: app/views/account/users/index.html.slim [hunk 5]
```diff
@@ -10,8 +10,12 @@
           .breadcrumb-wrapper.col-12
             = render_breadcrumbs({'Dashboard' => account_root_url,
                   'Users' => nil})
-      .content-header-right.text-md-right.col-md-6.col-12
-        = link_to 'New user', new_account_user_path, class: 'btn btn-primary'
+      .content-header-right.text-md-right.col-md-6.col-12.pr-0
+        .btn-group
+          .col-md-offset-2
```
Comment (igorkasyanchuk) [inline app/views/account/users/index.html.slim]: read about btngroup and how to use it

File: app/models/user.rb [hunk 6]
```diff
@@ -38,4 +38,12 @@ def super_admin?
   def privileged?
     admin? || super_admin?
   end
+
+  comma do
+    id
+    full_name
+    email
+    role
+    created_at
```
Comment (romaslmd) [inline app/models/user.rb]: will there be some date formatting?

File: app/views/account/users/index.html.slim [hunk 7]
```diff
@@ -11,7 +11,9 @@
             = render_breadcrumbs({'Dashboard' => account_root_url,
                   'Users' => nil})
       .content-header-right.text-md-right.col-md-6.col-12
-        = link_to 'New user', new_account_user_path, class: 'btn btn-primary'
+        .btn-group
+            = link_to 'Add new', new_account_user_path, class: 'btn btn-blue'
```
Comment (romaslmd) [inline app/views/account/users/index.html.slim]: is't there too many spaces? each lvl shound be in 2 spaces

File: app/controllers/account/users_controller.rb [hunk 8]
```diff
@@ -58,6 +58,10 @@ def destroy
     redirect_to account_users_path, flash: { success: 'User deleted!' }
   end
 
+  def download
+    send_data collection.to_comma, filename: "Users #{Date.today}.csv", disposition: 'attachment'
```
Comment (igorkasyanchuk) [inline app/controllers/account/users_controller.rb]: specs

File: app/models/user.rb [hunk 9]
```diff
@@ -38,4 +38,16 @@ def super_admin?
   def privileged?
     admin? || super_admin?
   end
+
+  def user_created
```
Comment (igorkasyanchuk) [inline app/models/user.rb]: remove this method
just return created_at as it's

File: spec/controllers/account/users_controller_spec.rb [hunk 10]
```diff
@@ -224,4 +224,14 @@
       expect(response).to redirect_to root_path
     end
   end
+
+  describe 'GET download' do
```
Comment (igorkasyanchuk) [inline spec/controllers/account/users_controller_spec.rb]: where are 3 additional tests?
without them PR can't be merged

File: spec/models/user_spec.rb [hunk 11]
```diff
@@ -10,4 +10,16 @@
   it 'user full name should be correct' do
     expect(user.full_name).to eq('Winston Churchill')
   end
+
+  it "admin? method should be correct" do
+    expect(user.admin?).to be false
+  end
+
+  it "super_admin? method should be correct" do
```
Comment (igorkasyanchuk) [inline spec/models/user_spec.rb]: all is good in PR, but 2 tests super_admin and admin are bad.
read our chat and find something else to test

Comment (igorkasyanchuk) [reply spec/models/user_spec.rb]: commented

**Pull request merged**